### PR TITLE
feat: new detailed evaluation interface

### PR DESCRIPTION
**Pull request by duyhungtnn**
Related https://github.com/bucketeer-io/bucketeer/issues/886
### Changes
- Added new type `BKTValue` to hold all valid types that could used on the interface. (refs from OpenFeature SDK interface)
```typescript
export type BKTPrimitiveValue = null | boolean | string | number
export type BKTJsonObject = {
  [key: string]: BKTValue
}
export type BKTJsonArray = BKTValue[]
/**
 * Represents a JSON node value.
 */
export type BKTValue = BKTPrimitiveValue | BKTJsonObject | BKTJsonArray
```
- Create the new data class `BKTEvaluationDetail<BKTValue>` to support getting variation details by type.
- Introduce a new interface to get variation details by variation type
```typescript
   objectVariation: (
    featureId: string,
    defaultValue: BKTValue,
  ): Promise<BKTValue>

   stringVariationDetails: (
    featureId: string,
    defaultValue: string,
  ):  Promise<BKTEvaluationDetail<string>>

  numberVariationDetails: (
    featureId: string,
    defaultValue: number,
  ):  Promise<BKTEvaluationDetail<number>>

  booleanVariationDetails: (
    featureId: string,
    defaultValue: boolean,
  ):  Promise<BKTEvaluationDetail<boolean>>
  
   objectVariationDetails: (
    featureId: string,
    defaultValue: BKTValue,
  ):  Promise<BKTEvaluationDetail<BKTValue>>
```

### Breaking changes
-  This PR introduces breaking changes on the method `getJsonVariation`.  The previous logic could return any type but we don't allow it anymore.
The returned value will be either a `BKTJsonObject` or a `BKTJsonArray`. If no result is found, it will return the provided `defaultValue`, which can be of any type within `BKTValue`. 

- Also this method `getJsonVariation` is marked as deprecated and the user should use `objectVariation` instead.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -116,6 +175,60 @@ export class BKTClientImpl implements Bucketeer {
     }, this.config.pollingIntervalForRegisterEvents!);
   }
 
+  async stringVariation(user: User, featureId: string, defaultValue: string): Promise<string> {
+    return (await this.stringVariationDetails(user, featureId, defaultValue)).variationValue;
+  }
```
Comment (cre8ivejp) [inline src/index.ts:180]: Can you move this right after the `booleanVariationDetails`?

File: src/objects/reason.ts [hunk 2]
```diff
@@ -3,11 +3,13 @@ export type Reason = {
...
+const ReasonTypeValue = [
+  'TARGET',
+  'RULE',
+  'DEFAULT',
+  'CLIENT',
+  'OFF_VARIATION',
+  'PREREQUISITE',
+] as const;
+
+export type ReasonType = (typeof ReasonTypeValue)[number];
```
[the remark this answers is not in the document]
Comment (duyhungtnn) [reply src/objects/reason.ts]: updated https://github.com/bucketeer-io/node-server-sdk/pull/70/commits/6fde52fc2b2a4a6ba78fdc5406f1cc66c61af923

File: src/index.ts (hunk 1 above)
Comment (duyhungtnn) [reply src/index.ts:180]: updated https://github.com/bucketeer-io/node-server-sdk/pull/70/commits/1884c252600c7182406c4f7630e96785702f2451

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**